lib/core/src/lib/pipes/full-name.pipe.spec.ts

```diff
@@ -16,42 +16,81 @@
  */
 
 import { FullNamePipe } from './full-name.pipe';
+import { UserLike } from './user-like.interface';
+import { cloneDeep } from 'lodash';
+
+interface TestCase {
+    title: string;
+    user: UserLike;
+    result: string;
+};
 
 describe('FullNamePipe', () => {
 
     let pipe: FullNamePipe;
 
-    beforeEach(() => {
-        pipe = new FullNamePipe();
-    });
+    const emptyUserTestCase: TestCase = {
+        title: 'should return empty string when there is no name',
+        user: {},
+        result: ''
+    };
 
-    it('should return empty string when there is no name', () => {
-        const user = {};
-        expect(pipe.transform(user)).toBe('');
-    });
+    const onlyUserEmailTestCase: TestCase = {
+        title: 'should return user email when firstName, lastName and username are not available',
+        user: { email: 'abcXyz@gmail.com' },
+        result: 'abcXyz@gmail.com'
+    };
 
-    it('should return only firstName as fullName when there is no lastName ', () => {
-        const user = {firstName : 'Abc'};
-        expect(pipe.transform(user)).toBe('Abc');
-    });
+    const baseTestCases: TestCase[] = [
+        {
+            title: 'should return only firstName as fullName when there is no lastName',
+            user: { firstName: 'Abc', lastName: '', username: '' },
+            result: 'Abc'
+        },
+        {
+            title: 'should return only lastName as fullName when there is no firstName',
+            user: { firstName: '', lastName: 'Xyz', username: '' },
+            result: 'Xyz'
+        },
+        {
+            title: 'should return fullName when firstName and lastName are available',
+            user: { firstName: 'Abc', lastName: 'Xyz', username: '' },
+            result: 'Abc Xyz'
+        },
+        {
+            title: 'should return username when firstName and lastName are not available',
+            user: { firstName: '', lastName: '', username: 'username' },
+            result: 'username'
+        }
+    ];
 
-    it('should return only lastName as fullName when there is no firstName ', () => {
-        const user = {lastName : 'Xyz'};
-        expect(pipe.transform(user)).toBe('Xyz');
-    });
+    [undefined, false, true].forEach(injectionToken => {
```

Comment:
can you please split that into traditional unit tests, so that other colleagues will be able to read that? honestly, this looks like scientific code, not readable for newcomers

Reply:
Sure, but I will get the duplication alert from Sonar for sure, but I think it is not a required check right?